File: use-cases/customer-growth-job-service/e2e/deploy/golang-sqlx-service/pkg/dto/customer_growth_job_dto.go
package dto

type CustomerGrowthJobSearchRequest struct {
	WorkspaceID string `json:"workspace_id"`
	ProjectID   string `json:"project_id"`
	Status      string `json:"status"`
	OwnerUserID string `json:"owner_user_id"`
}

type CreateCustomerGrowthJobRequest struct {
	ID          int64  `json:"id"`
	Region      string `json:"region"`
	TenantID    string `json:"tenant_id"`
	WorkspaceID string `json:"workspace_id"`
	ProjectID   string `json:"project_id"`
	JobID       string `json:"job_id"`
	DisplayName string `json:"display_name"`
	Status      string `json:"status"`
	OwnerUserID string `json:"owner_user_id"`
}

type UpdateCustomerGrowthJobRequest struct {
	DisplayName string `json:"display_name"`
	Status      string `json:"status"`
	OwnerUserID string `json:"owner_user_id"`
}

type CustomerGrowthJobDTO struct {
	ID          int64  `json:"id"`
	Region      string `json:"region"`
	TenantID    string `json:"tenant_id"`
	WorkspaceID string `json:"workspace_id"`
	ProjectID   string `json:"project_id"`
	JobID       string `json:"job_id"`
	DisplayName string `json:"display_name"`
	Status      string `json:"status"`
	OwnerUserID string `json:"owner_user_id"`
}

File: use-cases/customer-growth-job-service/e2e/deploy/python-sqlalchemy-service/app/dto/customer_growth_job_dto.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class CustomerGrowthJobSearchRequest:
    workspace_id: str | None = None
    project_id: str | None = None
    status: str | None = None
    owner_user_id: str | None = None


@dataclass(frozen=True)
class CreateCustomerGrowthJobRequest:
    id: int
    region: str
    tenant_id: str
    workspace_id: str
    project_id: str
    job_id: str
    display_name: str
    status: str
    owner_user_id: str


@dataclass(frozen=True)
class UpdateCustomerGrowthJobRequest:
    display_name: str
    status: str
    owner_user_id: str


@dataclass(frozen=True)
class CustomerGrowthJobDTO:
    id: int
    region: str
    tenant_id: str
    workspace_id: str
    project_id: str
    job_id: str
    display_name: str
    status: str
    owner_user_id: str
